static/js/sefaria/textCache.js

```diff
@@ -0,0 +1,75 @@
+class SegmentVersionsFofLanguage {
+    setVersion(versionTitle, version) {
+        this[versionTitle] = version;
+    }
+    updateVersion(segmentVersionsFofLanguage) {
+        Object.keys(segmentVersionsFofLanguage).forEach((versionTitle) => {
+            this.setVersion(versionTitle, segmentVersionsFofLanguage[versionTitle]);
+        });
+    }
+    getVersion(versionTitle) {
+        return this[versionTitle];
+    }
+}
+
+
+class SegmentVersions {
+    addLangIfMissing(language) {
+        if (!this.hasOwnProperty(language)) {
+            this[language] = new SegmentVersionsFofLanguage();
+        }
+    }
+    setVersion(language, versionTitle, version) {
+        this.addLangIfMissing(language);
+        this[language].setVersion(versionTitle, version);
+    }
+    updateVersions(segmentVersion) {
+        Object.keys(segmentVersion).forEach((language) => {
+            this.addLangIfMissing(language);
+            this[language].updateVersion(segmentVersion[language]);
+        });
+    }
+    getVersion (language, versionTitle) {
+        return this[language]?.getVersion(versionTitle);
+    }
+}
+
+
+class SectionVersions {
+    constructor() {
+        this.segments = [];
+    }
+    setVersion(segmentsVersionsArray) {
+        segmentsVersionsArray.forEach((segment, index) => {
+            if (this.segments[index]) {
+                this.segments[index].updateVersion(segment);
+            } else {
+                this.segments[index] = segment;
+            }
+        });
+    }
+    getVersion(language, versionTitle) {
+        return this.segments.map((segment) => {
+            segment.getVersion(language, versionTitle);
+        });
+    }
+}
+
+
+export default class TextCache {
+    setSegmentVersion(ref, language, versionTitle, version) {
+        if (!this.hasOwnProperty(ref)) {
```

Comment:
In general It's better to use `in` instead of `hasOwnProperty`: `hasOwnProperty` will only return true if the key is on the object itself, but will return false if the key is in the prototype. The `in` operator will return true if the property is on the object itself or anywhere in its prototype chain. However, in your case, you're working with instances where prototype properties would not likely be a concern, but it's still a good practice.